### PR TITLE
Add periphery tests

### PR DESCRIPTION
**Pull request by andreivladbrg**
Please refer to #979 notes.

### Changes

- add `core` dir in `test` 
- move all current tests in `test/core`
- add `periphery` dir `test` 
- move all [periphery](https://github.com/sablier-labs/v2-periphery/tree/main/test) tests in `test/periphery`
- add `core` and `periphery` remappings
- add periphery precompiles 
- add `deployProtocol` function in `Precompiles` contract
- include periphery bytecodes in `update_precompiles` shell script

**TODO** after everything is merged here: add `PROTOCOL` heaader in `Precompiles`

### CODE REVIEW

**Review by smol-ninja — COMMENTED** [review]
Wouldn't it be better to use `src/core` and `src/periphery` in tests than `core` and `periphery`? As a good standard to differentiate that its being imported from the `src` directory.

**Comment by smol-ninja** [comment]
We should also merge the `mock`, `utils` and Base.t.sol`. WDYT? If yes, @andreivladbrg Should we do it as a part of this PR or via a separate issue (after current PRs are merged)?

File: precompiles/Precompiles.sol [hunk 1]
```diff
@@ -22,7 +25,7 @@ contract Precompiles {
     uint256 public constant MAX_TRANCHE_COUNT = 500;
 
     /*//////////////////////////////////////////////////////////////////////////
-                                     BYTECODES
+                                        CORE
```
[the remark this answers is not in the document]
Comment (andreivladbrg) [reply precompiles/Precompiles.sol:28]: agree, feel free to make this change

also, keep in mind the TODO in the OP

**Comment by andreivladbrg** [comment]
> We should also merge the `mock`, `utils` and `Base.t.sol`. WDYT? If yes, @andreivladbrg Should we do it as a part of this PR or via a separate issue (after current PRs are merged)?

Please see the next PR: https://github.com/sablier-labs/v2-core/pull/981

This PR simply adds the tests here.

> If you look at the Base test file of periphery, its a bit confusing to not have explicitly src directory and just use core

Why so? You can see the remappings configuration to see that `core` points to `src/core/`. I prefer `core`, as tests are not meant to be imported externally, and are not a part of the package.

**Comment by smol-ninja** [comment]
> Why so? You can see the remappings configuration to see that core points to src/core/. I personally prefer core because it is shorter, also tests are not meant to be imported externally, and are not a part of the package

I understand its in remapping but it makes it easier to see where files are from and reduces overhead when navigating around. Also, imo its not an intuitive remapping.

Second argument is that, by using `src/core` and `src/periphery`, all the imports from the `src` directory can be put together in test files which otherwise will be separated based on alphabetical order.

> tests are not meant to be imported externally, and are not a part of the package

Yes and that's why absolute paths are used to import from the `src` directory.

**Comment by andreivladbrg** [comment]
answered here: https://github.com/sablier-labs/v2-core/pull/981#issuecomment-2255921776

**Review by smol-ninja — APPROVED** [review]

**Pull request merged**